packages/core/src/routes/interaction/verifications/mfa-payload-verification.ts

```diff
@@ -33,10 +39,42 @@ const verifyBindTotp = async (
   return { type, secret };
 };
 
+const findUserTotp = (
+  mfaVerifications: User['mfaVerifications']
+): MfaVerificationTotp | undefined =>
+  mfaVerifications.find((mfa): mfa is MfaVerificationTotp => mfa.type === MfaFactor.TOTP);
+
+const verifyTotp = async (
+  mfaVerifications: User['mfaVerifications'],
+  payload: TotpVerificationPayload
+): Promise<MfaVerificationResult> => {
+  const totp = findUserTotp(mfaVerifications);
+
+  // Can not found totp verification, this is an invalid request, throw invalid code error anyway for security reason
+  assertThat(totp, 'session.mfa.invalid_totp_code');
+
+  const { code } = payload;
+  const { key, id, type } = totp;
+
+  assertThat(validateTotpToken(key, code), 'session.mfa.invalid_totp_code');
+
+  return { type, id };
+};
+
 export async function bindMfaPayloadVerification(
   ctx: WithLogContext,
   bindMfaPayload: BindMfaPayload,
   interactionStorage: AnonymousInteractionResult
 ): Promise<BindMfa> {
   return verifyBindTotp(interactionStorage, bindMfaPayload, ctx);
 }
+
+export async function mfaVerificationPayloadVerification(
```

Comment:
align with the above bindMfaPayload naming convention `verifyMdaPayloadVerification`?

Reply:
updated